R/supervised-model-deployment.R

```diff
@@ -457,6 +459,64 @@ SupervisedModelDeployment <- R6Class("SupervisedModelDeployment",
     deploy = function() {
     },
     
+    # Get and write model metadata
+    getMetadata = function() {
+
+      # Initialize list of metadata items
+      metadata <- list()
+      
+      load(private$modelInfoFile)
+      sesInfo <- sessionInfo()
+      
+      # Model name (custom if provided, default otherwise)
+      metadata$model_name <- if (is.null(self$params$modelName)) "default" else self$params$modelName
+       
+      # Model algorithm
+      metadata$algorithm <- private$algorithmShortName
+      
+      # Datetime model last trained
+      metadata$model_last_developed <- file.mtime(private$fitObjFile)
+      
+      # Datetime predictions made
+      metadata$prediction_datetime <- Sys.time()
+      
+      # R version
+      metadata$r_version <- paste0(sesInfo$R.version$major, ".", sesInfo$R.version$minor)
+      
+      # healthcare.ai version
+      metadata$healthcareai_version <- sesInfo$otherPkgs$healthcareai$Version
+      
+      # Other packages
+      packages <- c(sesInfo$basePkgs, names(sesInfo$otherPkgs))
+      metadata$other_packages_loaded <- packages[packages != "healthcareai"]
```

Comment:
Do you want to attach package versions as well?

Reply:
Yes, excellent.